x/ccv/provider/keeper/partial_set_security.go

```diff
@@ -308,11 +308,62 @@ func (k Keeper) CanValidateChain(ctx sdk.Context, chainID string, providerAddr t
 			!k.IsDenylisted(ctx, chainID, providerAddr))
 }
 
+// FulfillsMinStake returns true if the validator `providerAddr` has enough stake to validate chain `chainID`
+// by checking its staked tokens against the minimum stake required to validate the chain.
+func (k Keeper) FulfillsMinStake(ctx sdk.Context, chainID string, providerAddr types.ProviderConsAddress) bool {
+	minStake, found := k.GetMinStake(ctx, chainID)
+	if !found {
+		return true
+	}
+
+	validator, err := k.stakingKeeper.GetValidatorByConsAddr(ctx, providerAddr.Address)
+	if err != nil {
+		return false
+	}
+
+	// validator has enough stake to validate the chain
+	return validator.GetTokens().GTE(math.NewIntFromUint64(minStake))
+}
+
 // ComputeNextValidators computes the validators for the upcoming epoch based on the currently `bondedValidators`.
-func (k Keeper) ComputeNextValidators(ctx sdk.Context, chainID string, bondedValidators []stakingtypes.Validator) []types.ConsumerValidator {
+func (k Keeper) ComputeNextValidators(ctx sdk.Context, chainID string, bondedValidators []stakingtypes.Validator) []types.ConsensusValidator {
+	// sort the bonded validators by number of staked tokens in descending order
+	sort.Slice(bondedValidators, func(i, j int) bool {
+		return bondedValidators[i].GetTokens().GT(bondedValidators[j].GetTokens())
+	})
+
+	// if inactive validators are not allowed, only consider the first `MaxProviderConsensusValidators` validators
+	// since those are the ones that participate in consensus
+	allowInactiveVals, found := k.GetAllowInactiveValidators(ctx, chainID)
+	if !allowInactiveVals || !found {
+		// only leave the first MaxProviderConsensusValidators bonded validators
+		maxProviderConsensusVals := k.GetMaxProviderConsensusValidators(ctx)
+		if len(bondedValidators) > int(maxProviderConsensusVals) {
+			bondedValidators = bondedValidators[:maxProviderConsensusVals]
+		}
+	}
+
+	// take only the first `MaxValidatorRank` many validators; others are not allowed to validate
+	maxRank, found := k.GetMaxValidatorRank(ctx, chainID)
+	if found && maxRank > 0 && int(maxRank) < len(bondedValidators) {
+		tmpValidators := bondedValidators[:maxRank]
+
+		// also include other validators that have the same number of tokens as the last validator in the list
+		for i := int(maxRank); i < len(bondedValidators); i++ {
+			// if the validator has the same number of tokens as the last validator in the list, include it
+			if bondedValidators[i].GetTokens().Equal(bondedValidators[maxRank-1].GetTokens()) {
+				tmpValidators = append(tmpValidators, bondedValidators[i])
+			} else {
+				// since validators are sorted, we can break if we get to a validator with less tokens
+				break
+			}
+		}
+		bondedValidators = tmpValidators
+	}
```

Comment:
Do we actually want this max validator rank? It adds a bunch of logic and I'm not sure what's the use case.

Reply:
I also think that minStake should be enough. I would like to avoid adding more params just for the sake of it without actually having a real need. It just complicates the code and the UX.

Reply:
I will remove it then. Agreed that this probably makes UX more confusing. We can add it back if someone asks for it